src/extension/wps/src/main/java/au/org/emii/gogoduck/download/Config.java

```diff
@@ -0,0 +1,62 @@
+package au.org.emii.gogoduck.download;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Configuration setting for the ParallelDownloadManager
+ */
+public class Config {
+    private final Path downloadDirectory;
+    private final long localStorageLimit;
+    private final int downloadAttempts;
+
+    private Config(ConfigBuilder builder) {
+        downloadDirectory = builder.downloadDirectory;
+        localStorageLimit = builder.localStorageLimit;
+        downloadAttempts = builder.downloadAttempts;
+    }
+
+    public Path getDownloadDirectory() {
+        return downloadDirectory;
+    }
+
+    public long getLocalStorageLimit() {
+        return localStorageLimit;
+    }
+
+    public int getDownloadAttempts() {
+        return downloadAttempts;
+    }
+
+    public static class ConfigBuilder {
+        private Path downloadDirectory;
+        private long localStorageLimit;
+        private int downloadAttempts;
+
+        public ConfigBuilder() {
+            downloadDirectory = Paths.get(System.getProperty("java.io.tmpdir"));
+            localStorageLimit = 100 * 1024 * 1024; // 100MiB
+            downloadAttempts = 3;
```

Comment:
Do you think it will be better to move this to config and access it via au.org.emii.core.Config.

```
          localStorageLimit = 100 * 1024 * 1024; // 100MiB
           downloadAttempts = 3;
```